devicetypes/qubino/qubino-flush-2-relay.src/qubino-flush-2-relay.groovy

```diff
@@ -0,0 +1,450 @@
+	log.debug "Multichannel command ${cmd}" + (ep ? " from endpoint $ep" : "")
+	if (cmd.commandClass == 0x6C && cmd.parameter.size >= 4) { // Supervision encapsulated Message
+		// Supervision header is 4 bytes long, two bytes dropped here are the latter two bytes of the supervision header
+		cmd.parameter = cmd.parameter.drop(2)
+		// Updated Command Class/Command now with the remaining bytes
+		cmd.commandClass = cmd.parameter[0]
+		cmd.command = cmd.parameter[1]
+		cmd.parameter = cmd.parameter.drop(2)
+	}
+	def encapsulatedCommand = cmd.encapsulatedCommand()
+	zwaveEvent(encapsulatedCommand, cmd.sourceEndPoint as Integer)
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.basicv1.BasicReport cmd, ep = null) {
+	log.debug "Basic ${cmd}" + (ep ? " from endpoint $ep" : "")
+	changeSwitch(ep, cmd)
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.switchbinaryv1.SwitchBinaryReport cmd, ep = null) {
+	log.debug "Binary ${cmd}" + (ep ? " from endpoint $ep" : "")
+	changeSwitch(ep, cmd)
+}
+
+private changeSwitch(endpoint, cmd) {
+	def value = cmd.value ? "on" : "off"
+	if (endpoint == 1) {
+		createEvent(name: "switch", value: value, isStateChange: true, descriptionText: "Switch ${endpoint} is ${value}")
+	} else if (endpoint) {
+		String childDni = "${device.deviceNetworkId}:$endpoint"
+		def child = childDevices.find { it.deviceNetworkId == childDni }
+		child?.sendEvent(name: "switch", value: value, isStateChange: true, descriptionText: "Switch ${endpoint} is ${value}")
+	}
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.meterv3.MeterReport cmd, ep = null) {
+	log.debug "Meter ${cmd}" + (ep ? " from endpoint $ep" : "")
+	if (ep == 1) {
+		createEvent(createMeterEventMap(cmd))
+	} else if (ep) {
+		String childDni = "${device.deviceNetworkId}:$ep"
+		def child = childDevices.find { it.deviceNetworkId == childDni }
+		child?.sendEvent(createMeterEventMap(cmd))
+	}
+}
+
+private createMeterEventMap(cmd) {
+	def eventMap = [:]
+	if (cmd.meterType == 1) {
+		if (cmd.scale == 0) {
+			eventMap.name = "energy"
+			eventMap.value = cmd.scaledMeterValue
+			eventMap.unit = "kWh"
+		} else if (cmd.scale == 2) {
+			eventMap.name = "power"
+			eventMap.value = Math.round(cmd.scaledMeterValue)
+			eventMap.unit = "W"
+		}
+	}
+	eventMap
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.sensormultilevelv5.SensorMultilevelReport cmd, ep = null) {
+	log.debug "SensorMultilevelReport ${cmd}" + (ep ? " from endpoint $ep" : "")
+	def result = []
+
+	def map = [:]
+	switch (cmd.sensorType) {
+		case 1:
+			map.name = "temperature"
+			def cmdScale = cmd.scale == 1 ? "F" : "C"
+			map.value = convertTemperatureIfNeeded(cmd.scaledSensorValue, cmdScale, cmd.precision)
+			map.unit = getTemperatureScale()
+			break
+		default:
+			map.descriptionText = cmd.toString()
+	}
+	def child = childDevices.find { it.deviceNetworkId == state.temperatureSensorDni }
+	if (!child) {
+		addChildTemperatureSensor()
```

Comment:
you need to be setting `child = ` here

Reply:
you could also refactor this function to return the child after it's created so you wouldn't have to search for it after